### PR TITLE
Refactor logic

### LINKED ISSUE #5: [FR]  Use PyTorch Lightning
**Is your feature request related to a problem? Please describe.**
Training and logging should be more concise and unified for every model. 

**Describe the solution you'd like**
Use PyTorch Lightning.

### LINKED ISSUE #6: [BUG] Batch size > 1 is not allowed
**Describe the bug**
`--batch_size > 1` is not allowed. It also mixes with the input's slice dimension.

**To Reproduce**
Train a `cirim` or `e2evn` or `unet` with `--batch_size > 1`.

**Expected behavior**
`--batch_size > 1` should be allowed. When training on a dataset with varying matrix sizes though it should be limited to 1.

**Environment**
Operating System: `ubuntu-latest` Python Version: `>= 3.9` PyTorch Version: `>= 1.9`

### LINKED ISSUE #7: [FR] Unify Fast Fourier Transform across the repo
**Is your feature request related to a problem? Please describe.**
Fast Fourier Transform type, dim, and normalization should be consistent across the repo.

**Describe the solution you'd like**
Define `fft_type`, `fft_dim`, and `fft_normalization` across the repo. 

- `fft_type` options should be `"orthogonal"` and `"backward"`.
- `fft_normalization` options should be `"orthogonal"`, `"backward"`, and `None`.
- `fft_dim` should always be the last two dimensions or the two last before the complex dimension [..., 2], when this exists.

### LINKED ISSUE #9: [FR] Create global arguments file and inherit them
**Is your feature request related to a problem? Please describe.**
Every train and run script duplicates global arguments. 

**Describe the solution you'd like**
Global arguments should be on a separate file and be inherited when required.

**Describe alternatives you've considered**
This might be also addressed through `PL` and `WANDB`.

### LINKED ISSUE #11: [FR] Parse arguments for train and run from yaml file
**Describe the solution you'd like**
Use yaml files to parse arguments for training and running methods.

### LINKED ISSUE #16: [FR] Use weights and biases
**Is your feature request related to a problem? Please describe.**
Hyperparameters search options need. Logging needs to be refined.

**Describe the solution you'd like**
Use weights and biases.

### PR DESCRIPTION
**Pull request by wdika**
- Refactors logic.
- Improves backend using pytorch-lighting, hydra, and omegaconf.
- Addresses #5, #9, #11, #16.

### CODE REVIEW

**Review by wdika — COMMENTED** [review]
All changes have been checked at this point. Models' behavior needs to be verified before merging.
